test/unit/authentication/strategy/provider_oauth2_test.rb

```diff
@@ -193,10 +193,6 @@ class SsoSignupTest < ActiveSupport::TestCase
         assert authentication_strategy.error_message.blank?
         assert authentication_strategy.new_user_created?
       end
-
-      last_email = ActionMailer::Base.deliveries.last
-      assert_match 'Account Activation', last_email.subject
-      assert_match 'activate your account', last_email.body.to_s
```

Comment:
But why this is a fix? Why is not last email correct? Or should we check the first one? I think testing the email after signup is better than not.

Maybe it would be better to put a `teardown` to clear the deliveries for `SsoSignupTest` ?

Reply:
> But why this is a fix? Why is not last email correct? Or should we check the first one? I think testing the email after signup is better than not.Ç

The PR description should answer the question.
TD;DR; the email should not be sent in this case (when the user's `email_verified == true`)

> Maybe it would be better to put a teardown to clear the deliveries for SsoSignupTest ?

That could be an extra measure.

Reply:
my bad, but also shouldn't we add deliveries clearing teardown for the other test that created this in the first place?

Reply:
Added https://github.com/3scale/porta/pull/3629/commits/3906676ed8a902df79fc91b8ca5eb4886f9b1daf

Reply:
We discussed this with @jlledom [here](https://github.com/3scale/porta/pull/3629#issuecomment-1819011978), and it could be done in another PR. Basically, as I think it might cause so extra issues, also would require cleaning up individual deliveries cleanup (sorry for bad wording :grimacing: )

Reply:
I looked a little bit more careful at this. I think every test class that deals with mail deliveries should ensure they are properly cleared before and after its tests. So I added a commit to implement this for this particular one. If we spot other such classes we can add this in the same way.

Also I'm open to other ideas but this is what I came up with.